api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -186,6 +186,23 @@ type AzureManagedControlPlaneSpec struct {
 	// For authentication with Azure Container Registry.
 	// +optional
 	KubeletUserAssignedIdentity string `json:"kubeletUserAssignedIdentity,omitempty"`
+
+	// HTTPProxyConfig is the HTTP proxy configuration for the cluster.
+	// Immutable.
+	// +optional
+	HTTPProxyConfig *HTTPProxyConfig `json:"httpProxyConfig,omitempty"`
+}
+
+// HTTPProxyConfig is the HTTP proxy configuration for the cluster.
+type HTTPProxyConfig struct {
+	// HTTPProxy - The HTTP proxy server endpoint to use.
+	HTTPProxy *string `json:"httpProxy,omitempty"`
```

Comment:
Does this need to be a pointer, or would an empty string suffice to show that it's not set?

Reply:
I think it would make sense to have this as a pointer, as it will be in parallel to what the `containerservice` module expects ?

Reply:
That makes sense, I just wonder if we're using too many pointers tbh.

Reply:
^ Jont828 asked a good question.

Questions for @mjnovice, do the empty fields, i.e. `""` mean anything significant for the fields of `HTTPProxyConfig` ? 
It makes sense to me that `HTTPProxyConfig *HTTPProxyConfig json:"httpProxyConfig,omitempty"` is a pointer. `httpProxyConfig` does not have a in-built `nil` type and we want to distinguish it between having it assigned vs leaving it empty. 

However, I believe the fields of `HTTPProxyConfig` can be defined as non-pointers unless we want to explicitly differentiate between `""` and it not being set.

This article was quite handy for me to understand if the field should be a pointer type vs a non-pointer type. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md

Reply:
I also ran a simple test to check if it could cause any problems. 
https://goplay.tools/snippet/IJM-wX1BDdR

Reply:
@nawazkh you both raise valid points, however I was just following a pattern which we already have in the file. For e.g https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/managedclusters/spec.go#L167 , didn't want to create an anti-pattern. 

I we want to move away from pointers, that could be a refactor PR ?

Reply:
Here is the k8s API convention docs on optional vs required fields: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

specifically:

```
Optional fields have the following properties:

- They have the +optional comment tag in Go.
- They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices).
- The API server should allow POSTing and PUTing a resource with this field unset.
```

so all the optional fields here should be pointer types and have the `+optional` comment

Reply:
@CecileRobertMichon except for the slice I believe

Reply:
correct